pkg/controller/controller_utils.go

```diff
@@ -956,12 +955,37 @@ func FilterActivePods(pods []*v1.Pod) []*v1.Pod {
 	return result
 }
 
+func FilterTerminatingPods(pods []*v1.Pod) []*v1.Pod {
+	var result []*v1.Pod
+	for _, p := range pods {
+		if IsPodTerminating(p) {
+			result = append(result, p)
+		}
+	}
+	return result
+}
+
+func CountTerminatingPods(pods []*v1.Pod) int32 {
+	numberOfTerminatingPods := 0
```

Comment:
I think implementing `CountTerminatingPods` is a performance optimization not to allocate memory when not needed. ie. when TerminatingOrFailed is used. Though, I'm not sure it is worth the complication, but I'm fine with it. wdyt @alculquicondor ?

Reply:
Yea that was my thought.  We don't need to create a list if we don't need to track it.

Reply:
I updated the PR to call this function only when needed.  We do end up using this list in one place so I'd prefer to leave this function in place.

Reply:
continuing discussion in https://github.com/kubernetes/kubernetes/pull/117015#discussion_r1265582404

Reply:
We would actually benefit more from having a `ForEachTerminatingPod` method (to avoid storing a list that we ultimately don't use). But we try to do this in beta, based on benchmarking.